extension/encodingextension/config.go

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package encodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encodingextension"
+
+import (
+	codec "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/codec"
+)
+
+type Config struct {
+	LogCodecs    map[string]codec.Log
+	MetricCodecs map[string]codec.Metric
+	TraceCodecs  map[string]codec.Trace
```

Comment:
@MovieStoreGuy @atoulme 

This would require all the codecs to be added in this repo. I'm a bit worried about additional burden on the maintainers.

What if we have the encoding/decoding extensions as a type of extensions similar to the storage extension?

Each encoding/decoding extension would require to implement some API and carry the codec inside of it instead of exposing it as an additional public API?

In that case, we would have `jaegerencoder` and `jaegerdecoder` extensions and others implementing already existing `traces.Marshal` and `traces.Unmarshal`.

Users could implement their own encoding/decoding extension outside of this repo.

WDYT?

Reply:
I need to review again how we work with the storage extension, I think that's a good idea in terms of UX.

Reply:
Makes sense to me, it would definitely unblock development and make it easier to build your own.

Reply:
OK I have reimplemented with your suggestions in mind. PTAL